src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -883,6 +883,36 @@ int getSlotForFieldOrPropertyOrEvent(Symbol member)
             }
         }
 
+        private void EnforceNotNullIfNotNull(SyntaxNode? syntaxOpt, LocalState state, ImmutableHashSet<string> inputParamNames, NullableFlowState outputState, ParameterSymbol? outputParam)
+        {
+            if (inputParamNames.IsEmpty)
+            {
+                return;
+            }
+
+            foreach (var inputParam in this.MethodParameters)
+            {
+                if (inputParamNames.Contains(inputParam.Name)
+                    && outputState.MayBeNull()
```

Comment:
>outputState.MayBeNull() [](start = 23, length = 23)

Can we check this once, before the loop?

Reply:
Done